### PR TITLE
Reconcile Update.{get_title,beautify_title}()

### PR DESCRIPTION
**Pull request by StephenCoady**
As per https://github.com/fedora-infra/bodhi/issues/3114
Reconcile the two functions into one. The beautification now depends on
a boolean within get_title and otherwise nothing else changes.

Signed-off-by: Stephen Coady <scoady@redhat.com>

### CODE REVIEW

File: bodhi/server/models.py [hunk 1]
```diff
@@ -2583,13 +2584,41 @@ def get_title(self, delim=' ', limit=None, after_limit='…'):
...
+                return self.display_name
+
+            def build_label(build):
+                return build.nvr if nvr else build.package.name
+
+            if len(self.builds) > 2:
+                title = ", ".join([build_label(build) for build in self.builds[:2]])
+
+                if amp:
+                    title += ", &amp; "
```
Comment (pypingou) [inline bodhi/server/models.py]: This has become simply "&" since this PR was opened.

I'm curious why this isn't being caught by the CI system though :(

**Review by pypingou — DISMISSED** [review]

File: bodhi/server/models.py (hunk 1 above)
Comment (StephenCoady) [reply bodhi/server/models.py]: OK I'll make that change.

File: bodhi/server/models.py [hunk 2]
```diff
@@ -4106,7 +4100,7 @@ def update_summary(self):
             list: A list of dictionaries with keys 'alias' and 'title', indexing each update alias
                 and title associated with this Compose.
         """
-        return [{'alias': u.alias, 'title': u.beautify_title(nvr=True)} for u in self.updates]
+        return [{'alias': u.alias, 'title': u.get_title(nvr=True)} for u in self.updates]
```
Comment (nphilipp) [inline bodhi/server/models.py]: If `beautify=False` is the default, this should be `u.get_title(nvr=True, beautify=True)` instead.

File: bodhi/server/templates/fragments.html [hunk 3]
```diff
@@ -182,8 +182,8 @@
                   ${util.type2icon(update['type'])|n}
                   <div class="media-body ml-2">
                       <div class="font-weight-bold">
-                        <a href="${request.route_url('update', id=update['alias'])}">
-                        ${update.beautify_title(amp=True,nvr=True) | h}
+                        <a href="${request.route_url('update', id=update['alias'])}"">
```
Comment (nphilipp) [inline bodhi/server/templates/fragments.html]: That double double-quote at the end looks dubious.

File: bodhi/server/templates/fragments.html [hunk 4]
```diff
@@ -229,14 +229,14 @@
...
-            <small>created 
+            <small>created
               %if display_user:
               by
                 <a href="${request.route_url('user', name=override['submitter']['name'])}">
                   ${override['submitter']['name']}
                 </a>
               %endif
-              <span title="${override['submission_date']} UTC"> ${util.age(override['submission_date'], True)}</span> ago 
+              <span title="${override['submission_date']} UTC"> ${util.age(override['submission_date'], True)}</span> ago
```
Comment (nphilipp) [inline bodhi/server/templates/fragments.html]: Again, unrelated white space fixes.

File: bodhi/server/templates/fragments.html [hunk 5]
```diff
@@ -128,7 +128,7 @@
           </div>
       </div>
       </div>
-      
+
```
Comment (nphilipp) [inline bodhi/server/templates/fragments.html]: These look like unrelated white space fixes to me, not sure what the consensus is how we want to deal with these kind of fixes. @cverna @pypingou @abompard?

File: bodhi/server/templates/update.html [hunk 6]
```diff
@@ -57,13 +57,13 @@
             </div>
                 <div class="media-body ml-0">
                     <div class="font-weight-bold">
-                      <a href="${request.route_url('update', id=update['alias'])}">
-                      ${update.beautify_title(amp=True,nvr=True) | h}
+                      <a href="${request.route_url('update', id=update['alias'])}"">
```
Comment (nphilipp) [inline bodhi/server/templates/update.html]: Another double double-quote.

File: bodhi/server/templates/update.html [hunk 7]
```diff
@@ -1182,7 +1182,7 @@ <h3 class="modal-title" id="waiveModalLabel">Waive All Test Results</h3>
     }
 
     button.addEventListener('click', function(event) {
-        if (document.selection) { 
+        if (document.selection) {
```
Comment (nphilipp) [inline bodhi/server/templates/update.html]: More unrelated white space fixes.

**Review by nphilipp — CHANGES_REQUESTED** [review]

**Review by nphilipp — APPROVED** [review]
